### PR TITLE
feat(dialect): add the MySQL functions with no SQLite spelling

### PR DESCRIPTION
**Pull request by nao1215**
## What

The first half of #595: the MySQL scalar functions that had no SQLite spelling and no helper here, so a query using one was told "no such function" about a name the caller did write and MySQL does define. Five of the missing names were plain spellings of something this package already had -- `LCASE` and `UCASE` are `LOWER` and `UPPER`, `MID` is `SUBSTRING`, `DAYOFMONTH` is `DAY`, `REGEXP_LIKE` is the `REGEXP` operator -- so a query failed over which of two names for one function its author happened to type.

Added: `LCASE`, `UCASE`, `MID`, `DAYOFMONTH`, `ISNULL`, `STRCMP`, `BIT_LENGTH`, `INSERT`, `TO_BASE64`, `FROM_BASE64`, `CONV`, `BIN`, `OCT`, `BIT_COUNT`, `COT`, the two-argument `ATAN`, `CRC32`, `REGEXP_LIKE`, `REGEXP_SUBSTR`, `REGEXP_INSTR`, `SHA1`, `SHA2`, `INET_ATON`, `INET_NTOA`, `IS_IPV4` and `IS_IPV6`.

## How

The helpers live in a new `dialect/mysql_functions.go`, registered under the names MySQL gives them. A rewrite is only needed for a name SQLite already means something else by, and here that is three: `ISNULL` and `INSERT` are SQLite keywords, so the call form is a syntax error there whatever this package registers, and SQLite's `atan` takes one argument where MySQL's takes one or two. Leaving the other twenty-three as the query wrote them is not laziness -- a rewrite changes the SQL text, and the result column is named after that text, so a rewritten call needs its original name put back as an alias. A registration needs none of that.

`registerAll`'s two anonymous struct types became a named `scalarSpec` so the MySQL-only table could be a function in its own file rather than another two hundred lines in `functions.go`.

## Getting the edges right

Every expected value in the tests was read from a running MySQL 8.4.11 rather than derived, because these are the functions whose edges are easiest to guess wrong. The ones that are pinned as tests because guessing would have got them wrong:

- `CONV` reads unsigned and stops at the first character its base does not have, so `CONV('12abc', 10, 10)` is 12 and `CONV('xyz', 16, 10)` is 0 while `CONV('', 16, 10)` is NULL. It saturates rather than wrapping, so `CONV('ffffffffffffffffff', 16, 10)` is the largest unsigned 64-bit value. A negative base asks for the signed reading, which is visible only when the sign bit is set: `CONV(-15, 10, -2)` is `-1111` and `CONV(-15, 10, 16)` is the same bits as `FFFFFFFFFFFFFFF1`.
- `INSERT` leaves the string alone for a position outside it, replaces everything from the position onwards for a negative length or one that reaches past the end, counts characters rather than bytes, and rounds a fractional position rather than truncating it.
- `TO_BASE64` breaks its output into lines of 76 characters, which a caller comparing against MySQL's own output sees, and `FROM_BASE64` ignores the whitespace that puts back.
- `INET_ATON` reads a short address by filling the top bytes from the left, so `1.2` is 1.0.0.2 and not 0.0.1.2, and refuses a part above 255 or a trailing space. `IS_IPV4` accepts leading zeros and refuses anything that is not four parts; `IS_IPV6` refuses a bare dotted quad and an address carrying a zone.
- The three `REGEXP_` functions fold case by default, for the same reason the `REGEXP` operator now does, and take the same match_type argument. Positions count characters, so `REGEXP_INSTR('日本語です', 'です')` is 4.
- `SHA2` answers NULL for a width MySQL does not offer and reads 0 as 256.

`STRCMP` is the one approximation, and the godoc says so: it folds case, which MySQL's default collation does, and not accents, which that collation also does, so `STRCMP('é', 'e')` is 0 in MySQL and a difference here. Matching that would mean carrying the collation's weight tables.

`COT` is the one value that can differ in the last bit, because Go's tangent and the C library's round differently. Its test compares with a tolerance rather than pinning the digits, and the godoc records both values.

## Testing

`dialect/mysql_functions_test.go` holds the measured table, plus two things a table cannot check. `TestMySQLOnlyFunctionRelations` asserts the relations these functions promise rather than one more fixed value each -- `LCASE(x)` equals `LOWER(x)` and `MID(x, 2, 3)` equals `SUBSTRING(x, 2, 3)` over a set of inputs, so an alias cannot drift away from the function it is a name for, and `FROM_BASE64(TO_BASE64(x))`, `CONV(CONV(x, 10, 36), 36, 10)` and `INET_NTOA(INET_ATON(x))` round-trip, including over a base-64 string long enough to be wrapped.

Running the 141 expressions through MySQL 8.4 and through this checkout now agrees on 138. The three that differ are `COT(1)` and `COT(-1)` in the last bit, and one `TO_BASE64` case where only the batch client's own escaping of the newline differs.

`make test`, `make lint` and `make lint-staticcheck` are clean, `go test -race ./dialect/` passes, the cross-builds for windows/amd64 and darwin/arm64 pass, and sqly builds and tests green against this checkout with a local replace. Translation benchmarks are flat: MySQL 13.9 us against 14.0 us and PostgreSQL 10.2 us against 10.2 us, medians of nine, with identical allocations, which is what adding registry entries and three switch cases should cost.

## Not in this change

The date and time half of #595 -- `SEC_TO_TIME`, `TIME_TO_SEC`, `TO_DAYS`, `FROM_DAYS`, `MAKEDATE`, `MAKETIME`, `PERIOD_ADD`, `PERIOD_DIFF`, `TIME_FORMAT`, `ADDTIME`, `SUBTIME` and `MICROSECOND`. A MySQL TIME runs from -838:59:59 to 838:59:59 and carries a sign, so it is not a clock time and the datetime helpers already here cannot be reused for it; those functions need a value type of their own and belong in their own change. #595 stays open for them.

Refs #595

## Summary by CodeRabbit

- **New Features**
  - Added support for numerous MySQL scalar functions, including string, encoding, numeric, cryptographic, regular expression, and IP-address helpers.
  - Added MySQL-compatible aliases, conversions, hashing formats, NULL handling, and input validation.
  - Improved compatibility for `INSERT`, `ISNULL`, and two-argument `ATAN` behavior.

- **Documentation**
  - Expanded documentation covering MySQL-specific functions and compatibility details.

- **Tests**
  - Added comprehensive compatibility tests for supported functions, edge cases, errors, aliases, and numeric behavior.

### CODE REVIEW

File: dialect/mysql_functions.go [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+		return src, nil
+	}
+	head := runes[:pos-1]
+	tail := len(runes)
+	// The sum is computed in int64 so a length near the top of the range cannot
+	// wrap around into a position inside the string.
+	if length >= 0 && pos-1+length < int64(len(runes)) {
+		tail = int(pos - 1 + length)
+	}
+	return string(head) + repl + string(runes[tail:]), nil
```
[the remark this answers is not in the document]
Comment (nao1215) [reply dialect/mysql_functions.go:135]: Fixed in f16f301 -- confirmed as a panic, not just a wrong answer: SELECT INSERT('abc', 2, 9223372036854775807, 'X') took the process down inside the SQLite callback, and MySQL 8.4 answers aX. The length is now bounded before the sum. The same class turned up twice more in this file and is fixed with it: BIT_LENGTH multiplies in int64 rather than int, and the REGEXP_ family refuses an occurrence past the number of characters, which cannot have a match and which also keeps the count inside what an int holds on a 32-bit build, where a truncated negative would have meant every match instead of none. Regression cases for all three are in TestMySQLOnlyFunctions.
